src/pure/formats.ml

```diff
@@ -53,24 +53,23 @@ let to_set_cookie
     ?expires ?max_age ?domain ?path ?secure ?http_only ?same_site name value =
 
   let expires =
-    match expires with
+    match Option.bind expires Ptime.of_float_s with
     | None -> ""
     | Some time ->
-      let time = Unix.gmtime time in
       let weekday =
-        match time.tm_wday with
-        | 0 -> "Sun" | 1 -> "Mon" | 2 -> "Tue" | 3 -> "Wed" | 4 -> "Thu"
-        | 5 -> "Fri" | 6 -> "Sat"
-        | _ -> assert false
+        match Ptime.weekday time with
+        | `Sun -> "Sun" | `Mon -> "Mon" | `Tue -> "Tue" | `Wed -> "Wed" | `Thu -> "Thu"
+        | `Fri -> "Fri" | `Sat -> "Sat"
       in
+      let ((y, m, d), ((hh, mm, ss), _tz_offset_s)) = Ptime.to_date_time time in
       let month =
-        match time.tm_mon with
+        match m with
         | 0 -> "Jan" | 1 -> "Feb" | 2 -> "Mar" | 3 -> "Apr" | 4 -> "May"
```

Comment:
Per the [docs of `Unix.tm`](https://ocaml.org/api/Unix.html#TYPEtm), `tm_mon` has range 0..11. By contrast, per the docs of [`Ptime.date`](https://erratique.ch/software/ptime/doc/Ptime.html#TYPEdate), the month field has range 1..12. Could you check this pattern?

Reply:
There's a good chance this is what is causing the [assertion failure in CI](https://github.com/aantron/dream/pull/100/checks#step:6:85).